bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/storage/ldb/KeyValueStorageRocksDB.java

```diff
@@ -0,0 +1,541 @@
+     * fresh and build a DB with the new settings (SSTs sizes, levels, etc..)
+     */
+    private static void doUpgradeIfNeeded(String path, DbConfigType dbConfigType, ServerConfiguration conf)
+            throws IOException {
+        FileSystem fileSystem = FileSystems.getDefault();
+        final Path rocksDbMarkerFile = fileSystem.getPath(path, ROCKSDB_MARKER);
+
+        if (Files.exists(fileSystem.getPath(path))) {
+            // Database already existing
+            if (Files.exists(rocksDbMarkerFile)) {
+                // Database was already created with RocksDB
+                return;
+            }
+        } else {
+            // Database not existing, no need to convert
+            return;
+        }
+
+        log.info("Converting existing database to RocksDB: {}", path);
+        long startTime = System.nanoTime();
+
+        String rocksDbPath = path + ".rocksdb";
+        KeyValueStorage source = new KeyValueStorageRocksDB(path, dbConfigType, conf, true /* read-only */);
+
+        log.info("Opened existing db, starting copy");
+        KeyValueStorage target = new KeyValueStorageRocksDB(rocksDbPath, dbConfigType, conf);
+
+        // Copy into new database. Write in batches to speed up the insertion
+        CloseableIterator<Entry<byte[], byte[]>> iterator = source.iterator();
+        Batch batch = target.newBatch();
+        try {
+            final int maxBatchSize = 10000;
+            int currentBatchSize = 0;
+
+            while (iterator.hasNext()) {
+                Entry<byte[], byte[]> entry = iterator.next();
+
+                batch.put(entry.getKey(), entry.getValue());
+                if (++currentBatchSize == maxBatchSize) {
+                    batch.flush();
+                    batch.clear();
+                    currentBatchSize = 0;
+                }
+            }
+
+            batch.flush();
+        } finally {
+            batch.close();
+            iterator.close();
+            source.close();
+            target.close();
+        }
+
+        FileUtils.deleteDirectory(new File(path));
+        Files.move(fileSystem.getPath(rocksDbPath), fileSystem.getPath(path));
+
+        // Create the marked to avoid conversion next time
+        Files.createFile(rocksDbMarkerFile);
+
+        log.info("Database conversion done. Total time: {}",
+                DurationFormatUtils.formatDurationHMS(TimeUnit.NANOSECONDS.toMillis(System.nanoTime() - startTime)));
+    }
+
+    private static void doCreateRocksDbMarker(String path) throws IOException {
+        try {
+            Files.createFile(FileSystems.getDefault().getPath(path, ROCKSDB_MARKER));
+        } catch (FileAlreadyExistsException e) {
+            // Ignore
+        }
+    }
+
+    private static final Comparator<byte[]> ByteComparator = UnsignedBytes.lexicographicalComparator();
+
+    private static final Logger log = LoggerFactory.getLogger(KeyValueStorageRocksDB.class);
```

Comment:
Does check style pass? I thought it wants constant at the beginning of the class

Reply:
Yes, I ran checkstyle & findbugs locally.